src/services/loginAndRegisterUser.ts

```diff
@@ -0,0 +1,21 @@
+import { auth } from "../firebase";
+import { loginUser, setLoading } from "../reducers/userSlice";
+import { AppDispatch } from "../types/types";
+
+export const loginAndRegisterUser = (dispatch: AppDispatch) => {
```

Comment:
Name of this function suggest me that it does login and register and the same time. Doesn't make sense imho. + it doesnt login or register "directly" - it registers function which is logic that happen after auth state of firebase API changes (as its own name `onAuthStateChanged` suggest that clearly). So this function is rather to register handler for authed user. Some name like: `registerAuthChangeHandler` will be more accurate for what this function actually does.